.github/workflows/ngsa.yaml

```diff
@@ -35,15 +35,40 @@ jobs:
 
     - name: Docker Build
       run: |
-        docker build . -t image
+        docker build . --progress auto -t image
+
+    - name: Docker Run and Test
+      run: |
+           
+            docker run --rm -d --name ngsacs -p 8080:8080 image --in-memory
+
+            echo "Waiting for web server to start ..."
+        
+            wait_time=10        
+            sleep $wait_time
+
+            # wait up to 30 seconds for web server to start
+            while true
+            do
+                if curl localhost:8080/version ; then
+                    echo -e "\n web server is running"
+                    break
+                fi
+                if [ $wait_time -gt 30 ] ; then
+                  echo -e "\n timeout waiting for web server to start"
+                  exit 1
+                fi
+                
+                sleep 1
+                ((wait_time=wait_time+1))
+            done
+
+            docker run --rm --network=host ghcr.io/retaildevcrews/ngsa-lr:beta -s "http://localhost:8080" --max-errors 1 -f baseline.json
```

Comment:
I have the same concern on Java that we need to rethink running the test on push.  I've posted a note that I'll convert to an issue.  We need to consider if we should run the test at all, run it always, or run it only when triggered by the cron job.